Show file header in the GUI, remove unused code

MBartkowiakSTFC opened the pull request:
**Description of work**
The `HDFTrajectoryInputData` class appeared to be an unnecessary layer between the `Trajectory` class and the rest of the code using trajectories. This PR removes it, together with some other classes that do not seem to be needed for anything.

closes #695 
closes #730 
closes #807
closes #811

**Fixes**
1. Removed all classes from `MDANSE.IO` except for `MinimalPDBReader`.
2. Removed all classes from `MDANSE.Framework.InputData`.
3. Replaced different `Configurator.get_information` implementations with a common `IConfigurator.__str__` implementation.
4. Adjusted the GUI code to use `Trajectory` instead of `HDFTrajectoryInputData`.
5. Introduced a common `check_metadata` function to `MDANSE.Framework.Formats.HDFFormat`, which is now used in the GUI for both trajectory and output data files.

**To test**
All the trajectory information displayed in the GUI should still be shown in the trajectory tab.
The plot selection tab should show additional information about the inputs of its analysis run. The 'metadata' section should now be followed by 'header' section.

MBartkowiakSTFC committed:
Remove unused or unnecessary classes

MBartkowiakSTFC committed:
Correct the unit tests

MBartkowiakSTFC committed:
Correct the trajectory in MolecularViewer

MBartkowiakSTFC committed:
Improve metadata and header view in the GUI

oerc0122 commented on MDANSE/Src/MDANSE/Framework/Configurators/IConfigurator.py:
```diff
@@ -145,6 +154,16 @@ def __init__(self, name, **kwargs):
         self._original_input = ""
 
+    def __str__(self) -> str:
+        return "\n".join(
+            [""]
+            + [
+                f"{label}={str(getattr(self, label, 'Not set'))}"
+                for label in self._printable_attributes
+            ]
+            + [f"{key}={str(self.get(key, 'Not set'))}" for key in self.keys()]
```
Maybe a usecase for [`value_chain`](https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.value_chain) is this more readable?
```suggestion
        return "\n".join(
            value_chain(
                "",
                (
                    f"{label}={str(getattr(self, label, 'Not set'))}"
                    for label in self._printable_attributes
                ),
                (f"{key}={str(self.get(key, 'Not set'))}" for key in self),
            )
        )
```

oerc0122 commented on MDANSE_GUI/Src/MDANSE_GUI/Tabs/Visualisers/TrajectoryInfo.py:
```diff
@@ -20,10 +20,68 @@
+    for molname, mollist in traj.chemical_system._clusters.items():
+        val.append(f"Molecule: {molname}; Count: {len(mollist)}")
+
+    for k, v in mol_types.items():
```
`mol_types` is empty?

oerc0122 commented on MDANSE/Src/MDANSE/Framework/Formats/HDFFormat.py:
```diff
@@ -28,6 +29,56 @@
+    """
+    meta_dict = {}
+
+    def put_into_dict(name, obj):
```
Needs type-hints and a docstring. Not sure what this is trying to do.

oerc0122 reviewed (changes requested):
Looks like a huge improvement in legibility and maintenance. Just a couple of minor things

MBartkowiakSTFC committed:
Add changes from code review, improve metadata output

MBartkowiakSTFC commented:
I implemented the changes you recommended.

Also, I noticed that TrajectoryEditor and CentreOfMassesTrajectory were not writing the 'metadata' group into the output file, since that was normally done by a method of `Converter`. Following the reasoning of this PR, I moved `write_metadata` to `MDANSE.Framework.Formats.HDFFormat` so it can be used by converters and other jobs that write trajectories. (This will be relevant to the TrajectoryFilter as well, once merged.)

oerc0122 commented on MDANSE/Src/MDANSE/Framework/Configurators/IConfigurator.py:
```diff
@@ -145,6 +159,18 @@ def __init__(self, name, **kwargs):
+        return "\n".join(
+            value_chain(
+                "",
+                [
+                    f"{label}={str(getattr(self, label, 'Not set'))}"
+                    for label in self._printable_attributes
+                ],
+                [f"{key}={str(self.get(key, 'Not set'))}" for key in self.keys()],
```
Shouldn't need to instantiate lists here, also, if this derives from `dict`, `keys` is the default iterable
```suggestion
                (
                    f"{label}={str(getattr(self, label, 'Not set'))}"
                    for label in self._printable_attributes
                ),
                (f"{key}={str(self.get(key, 'Not set'))}" for key in self),
```

oerc0122 commented on MDANSE/Src/MDANSE/Framework/Formats/HDFFormat.py:
```diff
@@ -28,6 +29,103 @@
+    return meta_dict
+
+
+def write_metadata(job: "IJob", output_file: h5py.File):
```
This seems like it might want to be a method on `IJob`, though I'm not 100% sure since it makes sense here too...

oerc0122 reviewed (approved):
Just a stylistic thing which may not be correct. 

Otherwise I'm happy.

MBartkowiakSTFC committed:
Remove unnecessary lists from IConfigurator.__str__
